src/pages/signin/GithubUsernameForm.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {Text, TextInput, View} from 'react-native';
+import styles from '../../styles/styles';
+import SubmitButton from './SubmitButton';
+import {setGitHubUsername} from '../../libs/actions/Session';
+
+class GithubUsernameForm extends React.Component {
+    constructor() {
+        super();
+
+        this.validateAndSubmitForm = this.validateAndSubmitForm.bind(this);
+
+        this.state = {
+            formError: false,
+            githubUsername: '',
+            isLoading: false,
+        };
+    }
+
+    /**
+     * Check that all the form fields are valid, then trigger the submit callback
+     */
+    validateAndSubmitForm() {
+        if (!this.state.githubUsername.trim()) {
+            this.setState({formError: 'Please fill out all fields'});
+            return;
+        }
+
+        this.setState({
+            formError: null,
+            isLoading: true,
```

Comment:
Same comment here. The loading state only gets set one way since we are banking on a redirect and it's not clear what if anything would unset it.